src/data/markdown/translated-guides/en/03 Using k6 browser/04 Migrating to k6 v0-46.md

```diff
@@ -0,0 +1,249 @@
+        browser: {
+          type: 'chromium',
+        },
+      },
+    },
+  },
+  thresholds: {
+    checks: ["rate==1.0"]
+  }
+}
+```
+
+</CodeGroup>
+
+
+
+
+## Opening a new page
+
+You can open a new page by using the imported [browser](/javascript-api/k6-experimental/browser/#browser-module-api) object.
+
+<CodeGroup labels={["After"]} lineNumbers={[true]}>
+
+<!-- eslint-skip -->
+
+```javascript
+export default async function () {
+    const page = browser.newPage();
+    // ...
+    page.close();
+}
+```
+</CodeGroup>
+
+<Blockquote mod="note" title="">
+
+Closing of the page is critical for the calculation of accurate Web Vital metrics. See the [browser metrics](/using-k6-browser/browser-metrics/) for more details.
+
+</Blockquote>
+
+
+
+
+## Simplified resource management
+
+As [mentioned earlier](#scenario-options), in the new API, there's a shift in how the browser's lifecycle is managed. Previously, users were required to handle the creation and closing of the browser instance using methods like `chromium.launch()` and `chromium.connect()` for creation, and the `browser.close()` method for releasing resources before ending the iteration. This repetitive code has been eliminated. Now, all that is needed is to specify the browser type within the [scenario options](#scenario-options). A browser instance will be automatically created and closed for each iteration, streamlining the process.
+
+Since the browser lifecycle is automatically managed by the browser module, the closing of the browser has been simplified. The explicit `browser.close()` call has been removed. Simply close the page using the [page.close()](/javascript-api/k6-experimental/browser/page/close/) method as in the example below.
+
+<CodeGroup labels={["Before"]} lineNumbers={[true]}>
+
+<!-- eslint-skip -->
+
+```javascript
+page.close();
+browser.close();
+```
+
+</CodeGroup>
+
+<CodeGroup labels={["After"]} lineNumbers={[true]}>
+
+<!-- eslint-skip -->
+
+```javascript
+page.close();
+```
+
+</CodeGroup>
+
+
+
+
+## Browser context limit
+
+Since the browser lifecycle is now being managed by the [browser module](/javascript-api/k6-experimental/browser/), the new browser implementation limits the use to a single active [BrowserContext](/javascript-api/k6-experimental/browser/browsercontext/) per iteration. This change supports better prediction of resource requirements for a test run and allows for more controlled management of [BrowserContext](/javascript-api/k6-experimental/browser/browsercontext/)s.
+
+If a new [BrowserContext](/javascript-api/k6-experimental/browser/browsercontext/) needs to be created, the existing one must be closed first using the [browserContext.close()](/javascript-api/k6-experimental/browser/browsercontext/close) method. And, a new one can be created either with the [browser.newContext()](/javascript-api/k6-experimental/browser/newcontext/) or [browser.newPage()](/javascript-api/k6-experimental/browser/newpage) methods.
```

Comment:
Maybe we can show a code snippet here on how to create a new BrowserContext?

Reply:
There are examples in the links. I'll write "check out the examples for creating a browsercontext".

Reply:
Update: Provided an example.